### PR TITLE
Compilation output

### PR DESCRIPTION
**Pull request by avesloguzova**

### CODE REVIEW

**Comment by avesloguzova** [comment]
I updated pull request

**Comment by deadok22** [comment]
Looks good. 

Please, add nullability annotations, rename instance fields, and add/remove empty lines according to our code style.

**Comment by avesloguzova** [comment]
I updated pull request.

File: jps-plugin/src/org/intellij/erlang/jps/rebar/RebarProcessAdapter.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        return BuildMessage.Kind.INFO;
+      case STATISTICS:
+        return BuildMessage.Kind.PROGRESS;
+      default:
+        throw new AssertionError();
+    }
+  }
+
+  private static boolean isMessageContinue(@NotNull String messagePart) {
+    return Character.isWhitespace(messagePart.charAt(0));
```
Comment (deadok22) [inline jps-plugin/src/org/intellij/erlang/jps/rebar/RebarProcessAdapter.java]: This is dangerous. You're not 100% sure you're not given an empty message here, yet you call charAt(0).

Comment (avesloguzova) [inline jps-plugin/src/org/intellij/erlang/jps/rebar/RebarProcessAdapter.java]: Ok

**Comment by avesloguzova** [comment]
Pull request was updated

File: jps-plugin/src/org/intellij/erlang/jps/builder/BuilderProcessAdapter.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  protected final String myBuilderName;
+  protected final String myCompileTargetRootPath;
+
+  protected BuilderProcessAdapter(@NotNull CompileContext context,
+                                  @NotNull String builderName,
+                                  @NotNull String compileTargetRootPath) {
+    myContext = context;
+    myBuilderName = builderName;
+    myCompileTargetRootPath = compileTargetRootPath;
+  }
```
Comment (deadok22) [inline jps-plugin/src/org/intellij/erlang/jps/builder/BuilderProcessAdapter.java]: A newline.

**Comment by deadok22** [comment]
The code is good. I'd merge it right now. But there are some problems with functionality.

I've tried to build [erlang_localtime](https://github.com/dmitryme/erlang_localtime), and there were several error reports, although the build completed successfully. Please, try building open source projects, and ensure reports are processed correctly.

**Comment by avesloguzova** [comment]
Pull request was updated. I fix problem with erlang_localtime and other projects which use old rebar versions. Also I fix possible false error reports.

**Comment by deadok22** [comment]
It works better now.

There is one thing I don't like about rebar output processing. We get a lot of these messages produced by '-vv' argument to rebar. I think, it's inconvenient to see all of these messages in make output. Can we filter some of these messages out?

**Comment by avesloguzova** [comment]
Pull request was updated.

File: jps-plugin/src/org/intellij/erlang/jps/rebar/RebarProcessAdapter.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        return BuildMessage.Kind.INFO;
+      case STATISTICS:
+        return BuildMessage.Kind.PROGRESS;
+      default:
+        throw new AssertionError();
+    }
+  }
+
+  private static boolean isMessageContinue(@NotNull String messagePart) {
+    return messagePart.isEmpty()||Character.isWhitespace(messagePart.charAt(0));
```
Comment (deadok22) [inline jps-plugin/src/org/intellij/erlang/jps/rebar/RebarProcessAdapter.java]: Formatting

File: jps-plugin/src/org/intellij/erlang/jps/builder/ErlangCompilerError.java [hunk 4]
```diff
@@ -27,50 +28,63 @@
...
   }
 
   public int getLine() {
-    return line;
+    return myLine;
   }
 
   @Nullable
-  public static ErlangCompilerError create(String rootPath, String erlcMessage) {
+  public static ErlangCompilerError create(String rootPath, @NotNull String erlcMessage) {
```
Comment (deadok22) [inline jps-plugin/src/org/intellij/erlang/jps/builder/ErlangCompilerError.java]: Nullability annotation.

**Comment by avesloguzova** [comment]
Pull request was updated.

**Comment by deadok22** [comment]
Why haven't you removed '-vv' argument for rebar? We can also provide a notification when we skip some directories if we haven't found rebar.config there, or we can try to run rebar there instead.

Let's add some clarity and let users know we're building sources for tests along with production. I mean, currently, we have erlc command line arguments shown, and the only difference is '-DTEST' which is somewhat hard to spot.

**Comment by avesloguzova** [comment]
Sorry, it's my fault with argument '-vv' for rebar. I removed it from my laptop yesterday and after forget pull this changes.

**Comment by deadok22** [comment]
Which brings us to another problem - you haven't tested what you're submitting in a pull request.

**Comment by avesloguzova** [comment]
Pull request was updated

File: jps-plugin/src/org/intellij/erlang/jps/builder/ErlangCompilerError.java [hunk 5]
```diff
@@ -27,50 +28,63 @@
...
-    assert path != null;
-    return new ErlangCompilerError(details, VfsUtilCore.pathToUrl(path), lineNumber, category);
+    return createCompilerError(path.getPath(), line, warning, details);
   }
 
+  @NotNull
   public CompilerMessageCategory getCategory() {
-    return category;
+    return myCategory;
+  }
```
Comment (deadok22) [inline jps-plugin/src/org/intellij/erlang/jps/builder/ErlangCompilerError.java]: Members ordering

File: jps-plugin/src/org/intellij/erlang/jps/rebar/RebarProcessAdapter.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+import com.intellij.openapi.util.Key;
+import org.intellij.erlang.jps.builder.BuilderProcessAdapter;
+import org.intellij.erlang.jps.builder.ErlangCompilerProcessAdapter;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.jps.incremental.CompileContext;
+import org.jetbrains.jps.incremental.messages.BuildMessage;
+import org.jetbrains.jps.incremental.messages.CompilerMessage;
+
+
+public class  RebarProcessAdapter extends BuilderProcessAdapter {
```
Comment (deadok22) [inline jps-plugin/src/org/intellij/erlang/jps/rebar/RebarProcessAdapter.java]: Formatting

**Pull request merged**